utils/create-module.js

```diff
@@ -0,0 +1,131 @@
+        const modName = line.split('canvas-kit-react-')[1].slice(0, -2);
+        acc[modName] = line;
+        return acc;
+      }, {});
+    const sorted = Object.keys(exportsAsObj)
+      .sort()
+      .reduce((acc, key) => {
+        acc.push(exportsAsObj[key]);
+        return acc;
+      }, []);
+    sorted.push(''); // Will result in EOF line
+    return sorted;
+  };
+
+  const indexPath = path.join(cwd, 'modules/_canvas-kit-react/index.ts');
+  const lines = fs.readFileSync(indexPath, 'utf8').split('\n');
+
+  // Add our module's export statement then sort
+  const exportStatement = `export * from '@workday/canvas-kit-react-${moduleName}';`;
+  lines.push(exportStatement);
+  const sortedExports = sortExports(lines);
+
+  // Place our canvas kit export at the top
+  sortedExports.unshift("export {default as canvas} from '@workday/canvas-kit-react-core';");
+  fs.writeFileSync(indexPath, sortedExports.join('\n'), 'utf8');
+};
+
+const addImportToSassIndex = (cwd, moduleName) => {
+  const sortImports = lines => {
+    const importsAsObj = lines
+      .filter(line => !line.includes('$wdc-system-icons-path'))
+      .filter(line => line.length > 0)
+      .reduce((acc, line) => {
+        // Split out module name and slice off trailing /index.scss';
+        const modName = line.split('canvas-kit-css-')[1].split("/index.scss';")[0];
+        acc[modName] = line;
+        return acc;
+      }, {});
+    const sorted = Object.keys(importsAsObj)
+      .sort()
+      .reduce((acc, key) => {
+        acc.push(importsAsObj[key]);
+        return acc;
+      }, []);
+    sorted.push(''); // Will result in EOF line
+    return sorted;
+  };
+
+  const indexPath = path.join(cwd, 'modules/_canvas-kit-css/index.scss');
+  const lines = fs.readFileSync(indexPath, 'utf8').split('\n');
+
+  // Add our module's import statement then sort
+  const importStatement = `@import '~@workday/canvas-kit-css-${moduleName}/index.scss';`;
+  lines.push(importStatement);
+  const sortedImports = sortImports(lines);
+
+  // Place our system icons path at the top
+  sortedImports.unshift(
+    "$wdc-system-icons-path: '../../node_modules/@workday/canvas-system-icons-web/dist/svg' !default;"
+  );
+  fs.writeFileSync(indexPath, sortedImports.join('\n'), 'utf8');
+};
+
+// Only allow this script to be ran off command line
+if (require.main === module) {
+  const cwd = process.cwd();
+  const moduleName = process.argv[2];
+  let wantsCss = false;
+  if (process.argv.length >= 3) {
+    const arg = process.argv[3];
+    if (arg.toLowerCase() === 'css') {
+      wantsCss = true;
+    }
+  }
+  addModuleAsDependency(cwd, moduleName, wantsCss);
+  addExportToIndex(cwd, moduleName);
+  if (wantsCss) {
+    // If we add the import by calling this, it throws error:
+    // error Couldn't find package "@workday/canvas-kit-css-alert@^3.0.0-alpha.2" required by "@workday/canvas-kit-css@3.0.0-alpha.2" on the "npm" registry.
+    addImportToSassIndex(cwd, moduleName);
```

Comment:
This is the area in question

Reply:
I think you want `addExportToIndex` for 'react' module type and `addExportToSassIndex` for 'css' module type.

Reply:
Hrm, I was replicating what's currently advised at:
https://github.com/Workday/canvas-kit/blob/master/CONTRIBUTING.md#css

I think for the CSS one it IS and `import` no?

Here's essentially what I'm writing in that method:

```js
  const importStatement = `@import '~@workday/canvas-kit-css-${moduleName}/index.scss';`;
```

The issue I was seeing was I believe the shell script does some sort of `yarn install` after I've added the import and then it can't find it cause it hasn't been published.

What's kind of funny as I type this is _rubber ducky_ realization that if I were to follow the current instructions as it's written today, the shell script would complete _first_, and _then_ I'd go manually add the import (and it wouldn't fail the `yarn install`) 🤣 

So, hrm, answering my own question I'm thinking I might have to do one of:

1. See if I can somehow pull out the CSS part in to it's own script that strategically doesn't get called until > the shell script pulls in the `yarn install` stuff.
1. If that doesn't work, I could just add that back to the read me and we're still "better" in terms of less steps

Lmk if that sounds correct, but otherwise I'll try to find some spare cycles to make these changes. Thanks again @NicholasBoll 🙏

Reply:
I suppose the `index.scss` only has imports where the `index.ts` has both imports and exports. The name `addImportToSassIndex` is fine.

Reply:
I rearranged some things and it works. I run the `yarn` command before invoking your JS script. This seems to follow all the necessary steps outlined in https://github.com/Workday/canvas-kit/blob/master/CONTRIBUTING.md#creating-a-module

The chicken/egg problem is the CI will fail unless that package is published, but that would be an issue without automating anything. This is a problem.

Reply:
@roblevintennis So, you'll have to hard-code `0.0.0` as the package version when writing the dependency to `_canvas-kit-css/package.json` and `_canvas-kit-react/package.json`. That version is special and `yarn` will pass. When we go to create a release, lerna will set the version number correctly for us.